### DISCUSSION TITLE
Why is Prim-RPC not as popular?

### DISCUSSION THREAD

**Discussion by kkimdev** [doseofted/prim-rpc#92] [General]
It looks like a pretty cool library, and was wondering why this is not as popular. What do you think @doseofted ?

**Comment by doseofted** [comment]
Thank you! I wondered for a while myself 😅

When I first published the library as open source, I was writing blog articles about it, posting to discussion boards, and sharing examples on Twitter. That push helped grow the library's reach a little bit, especially within the Nuxt/Vue community. After the launch of the library, I've not posted about the library as much so those who find the library usually find it from an old blog post or through my profiles online.

While the library itself isn't commercial, there's still a bit of "marketing" that goes into getting developers familiar/comfortable with a new library and that can take quite a bit of time. I think it also helps to have connections in the JavaScript ecosystem to help promote the library (especially on YouTube and X) but I'm not really involved in that online community. I think it also helps to anchor what you're promoting with something familiar and more popular than the thing being promoted. For example, the tRPC library initially promoted itself as an easier alternative to GraphQL but for JS developers. I tried to do this with Prim+RPC by likening the library's simplicity to "just using JavaScript" and creating integrations with other frameworks but looking back, I don't think that strategy worked well.

But I think ultimately, it takes a lot of time to grow a library: especially when that library is a major component of your app like RPC frameworks typically are. Developers will want to know that the library will be maintained, that other developers are using it, and that it's well-regarded by others. It's fairly difficult initially to check all of those boxes. But I do have plans to keep working on Prim+RPC myself even if it doesn't gain much traction as an open source project (as I find the library very helpful myself). I eventually hope to push out a stable version 1.0 and I may do another promotion once that's released.
